ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -709,16 +709,26 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
         curTip      = castPoint $ AF.headPoint curChain
         curHead     = AF.headAnchor curChain
 
-    -- Either frag extends loe or loe extends frag
-    --
-    -- PRECONDITION: @AF.withinFragmentBounds (AF.anchorPoint frag) loe@
+    -- Either frag extends loe or loe extends frag. The fragment is represented
+    -- by the current chain and a diff with that current chain. It is tempting
+    -- to only consider the suffix of the diff, but that would be incorrect,
+    -- because the diff might not intersect with the LoE fragment, because the
+    -- diff suffix is anchored somewhere on the current chain and LoE frag's tip
+    -- might be older than that anchor.
```

Comment:
> Also, it strikes me like it may truncate the current selection too

This in turn could truncate to the same point in the current chain for any candidates forking after the intersection with the LoE fragment.

Reply:
> it may truncate the current selection

I don't think that is right. It may truncate the candidate fragment that coincides with the current selection, indeed, but the current selection is a stateful thing that is only changed if we find a _better_ candidate, so `followsLoEFrag` will not affect it.

I will propose a new documentation comment.

Reply:
1fea816f55571d975eda868dc2b6a9fc58264a14